src/Features/Core/Portable/InlineHints/InlineParameterHint.cs

```diff
@@ -2,21 +2,20 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
-
 namespace Microsoft.CodeAnalysis.InlineHints
 {
     internal readonly struct InlineParameterHint
     {
-        public readonly SymbolKey ParameterSymbolKey;
-        public readonly string Name;
+        public readonly IParameterSymbol? Parameter;
         public readonly int Position;
         public readonly InlineParameterHintKind Kind;
 
-        public InlineParameterHint(SymbolKey parameterSymbolKey, string name, int position, InlineParameterHintKind kind)
+        public InlineParameterHint(
+            IParameterSymbol? parameter,
```

Comment:
When is this null?

Reply:
so, for this API i've actually made it so that the subclass can provide information about all arguments.  And then the higher level decides what to do.  This is important as we have certain behavior depending on if something id the "first" argument or not.  We don't want it to be the "First non-null matching arg", but really only the first arg.